compiler/rustc_trait_selection/src/solve/eval_ctxt.rs

```diff
@@ -548,4 +567,56 @@ impl<'tcx> EvalCtxt<'_, 'tcx> {
     pub(super) fn universe(&self) -> ty::UniverseIndex {
         self.infcx.universe()
     }
+
+    pub(super) fn create_next_universe(&self) -> ty::UniverseIndex {
```

Comment:
where do we create a new universe outside of canonicalization and when instantiating binders?